django/contrib/admin/views/main.py

```diff
@@ -371,7 +372,12 @@ def get_ordering_field(self, field_name):
             elif hasattr(self.model_admin, field_name):
                 attr = getattr(self.model_admin, field_name)
             else:
-                attr = getattr(self.model, field_name)
+                try:
+                    attr = getattr(self.model, field_name)
+                except AttributeError:
+                    if LOOKUP_SEP in field_name:
+                        return field_name
+                    raise
```

Comment:
The ordering itself has a test. However... this part to add the ordering buttons to the table headers does not, you are right. I need to dig around and see how these are tested elsewhere.

Reply:
Thanks for the test!

Though, now I'm puzzled: last review I'm pretty sure I checked that the `and LOOKUP_SEP not in field_name:` in `result_headers` was covered by a test (i.e. if I commented out that part of the if guard, a test would fail). But now I do that and no test fail :thinking:

So, for a change like this:
```diff
diff --git a/django/contrib/admin/templatetags/admin_list.py b/django/contrib/admin/templatetags/admin_list.py
index 5bb0200c92..cb171bf19c 100644
--- a/django/contrib/admin/templatetags/admin_list.py
+++ b/django/contrib/admin/templatetags/admin_list.py
@@ -113,7 +113,7 @@ def result_headers(cl):
             # Set ordering for attr that is a property, if defined.
             if isinstance(attr, property) and hasattr(attr, "fget"):
                 admin_order_field = getattr(attr.fget, "admin_order_field", None)
-            if not admin_order_field and LOOKUP_SEP not in field_name:
+            if not admin_order_field:  #  and LOOKUP_SEP not in field_name:
                 is_field_sortable = False
 
         if not is_field_sortable:
```

No test fail but of course the headers in the change list view do not allow ordering by the `__` columns using related fields. Any ideas?

In any case, I was wondering if we should move the changes related to exposing ordering via the headers (with its new? test) to a separated commit. Since that specific change is not crucial for the implementation of the feature, I figured we could have two related but separated commits to make each diff more approachable.

What do you think?

Reply:
This seems reasonable. I am not sure exactly why it's not covered, I thought the test I added would have done it, but maybe testing the default ordering isn't enough to also test the table headers.

It seems reasonable to me. To be clear: do you want me to rebase this PR into two commits, or do you also want to remove that second commit from this PR and handle it after?

Reply:
Yes, I do think that rebasing this PR into two commits is a good path forward, the first one with a commit message "Fixed #10743 -- ...."  and the second with "Refs #10743 -- ..."

Reply:
I've split into two commits. The second commit is very small however. It turns out you can comment out the entire line, and the one below it without breaking any tests. So I think there are already no tests around this. I'm not sure how I could test this other than with selenium, which will take a bit of time as it's been years since I've used it.

Reply:
> I've split into two commits. The second commit is very small however. It turns out you can comment out the entire line, and the one below it without breaking any tests. So I think there are already no tests around this. I'm not sure how I could test this other than with selenium, which will take a bit of time as it's been years since I've used it.

Thank you! And no worries. I'll see if I can add a test myself, I'll seat with this branch again next week, after the 5.0 rc1 release (next Monday!).

Reply:
Hey @knyghty, I circled back on this PR.

Would you have time to add the Selenium tests that we discussed last month, to test the ordering of the admin columns?

Reply:
I will be able to take a look, but it may be next month at this point

Reply:
Perfect, sounds great! Thank you.